cea/optimization/slave/least_cost.py

```diff
@@ -190,7 +190,7 @@ def source_activator(Q_therm_req_W, hour, context):
                     E_coldsource_HPSew_W = float(Q_HPSew_cold_primary_W)
 
                 if (
-                MS_Var.GHP_on) == 1 and hour >= MS_Var.GHP_SEASON_ON and hour <= MS_Var.GHP_SEASON_OFF and Q_therm_req_W > 0:
+                MS_Var.GHP_on) == 1 and hour >= MS_Var.GHP_SEASON_ON and hour <= MS_Var.GHP_SEASON_OFF and Q_therm_req_W > 0 and tdhsup_K != tdhret_req_K:
```

Comment:
when comparing floating point numbers, it is often unwise to use `!=` and `==`, as they could be very close but still not the same due to how floating point numbers are represented by the computer. You could use `not np.isclose(tdhsup_K, tdhret_req_K)`